approov-service/src/main/java/io/approov/service/okhttp/ApproovService.java

```diff
@@ -276,31 +276,54 @@ static synchronized String getBindingHeader() {
     }
 
     /**
-     * Sets the interceptor extensions callback handler. This facility was introduced to support
-     * message signing that is independent from the rest of the attestation flow. The default
-     * ApproovService layer issues no callbacks, provide a non-null ApproovInterceptorExtensions
-     * handler to add functionality to the attestation flow.
+     * Sets the ApproovServiceMutator instance to handle callbacks from the
+     * ApproovService implementation. This facility enables customization of
+     * ApproovService operations at key points in the configuration and
+     * attestation flows. It should reduce the number of times this service
+     * layer implementation needs to be forked in order to introduce custom
+     * behavior.
      *
-     * @param callbacks is the configuration used to control message signing. The behaviour of the
-     *              provided configuration must remain constant while in use by the ApproovService.
-     *              Passing null to this method will disable message signing.
+     * @param mutator is the ApproovServiceMutator with callback handlers that may
+     *                override the default behavior of the ApproovService singleton.
+     *                Passing null to this method will reinstate the default
+     *                behavior.
      */
-    public static synchronized void setApproovInterceptorExtensions(ApproovInterceptorExtensions callbacks) {
-        if (callbacks == null) {
-            Log.d(TAG, "Interceptor extension disabled");
+    public static synchronized void setServiceMutator(ApproovServiceMutator mutator) {
+        if (mutator == null) {
+            Log.d(TAG, "Applied default ApproovServiceMutator");
+            mutator = new ApproovServiceMutator() {};
         } else {
-            Log.d(TAG, "Interceptor extension enabled");
+            Log.d(TAG, "Applied custom ApproovServiceMutator:" + mutator.toString());
         }
-        interceptorExtensions = callbacks;
+        serviceMutator = mutator;
+    }
+
+    /**
+     * @deprecated Use setApproovServiceMutator instead
```

Comment:
```suggestion
     * @deprecated Use setServiceMutator instead
```

Alternatively, rename the function to match (but be aware that `getServiceMutator()` should be updated for consistency in this case).